### DISCUSSION TITLE
v7 static config - how do I localize screen names

### DISCUSSION THREAD

**Discussion by kferch-teamworks** [react-navigation/react-navigation#12156] [General]
I really like the static config API.  I'm fighting with how to translate the screen names.  Since it's static config it's created so early and no easy way to change locales without recreating everything.  Haven't tried dynamic or v6 but I assume translating screen names based on locale changes via global context works with those?  Any thoughts on if v7 static might ever work with locale context changes?

Thanks,

Kral

**Comment by kferch-teamworks** [comment]
I tried useNavigation() -> navigation.setOptions({ title: t("some-key") }); but get the error:

`Cannot update a component (`NativeStackNavigator`) while rendering a different component`

**Comment by ThunbergOlle** [comment]
Good question, I'm also interested in this!

**Comment by satya164** [comment]
Screen names aren't supposed to be localized. For header title, currently you can provide a custom header title component to translate them:

```js
import { HeaderTitle } from '@react-navigation/elements';

const TranslatedTitle = ({ allowFontScaling, tintColor, style, children }) => {
  const t = useTranslation();

  return (
    <HeaderTitle>
      {t(children)}
    </HeaderTitle>
  ),
};
```

And then:

```js
screenOptions: {
  headerTitle: (props) => <TranslatedTitle {...props} />
}
```

When React Native gets React 19, I'll look into supporting [`use` hook](https://react.dev/reference/react/use) so that context can be used in `options` directly.

**Reply by ThunbergOlle** [reply]
Thanks!

**Reply by itsramiel** [reply]
@satya164 react 19 is here now. Any updates. I would rather not pass an entire component but rather just the string

**Reply by satya164** [reply]
@itsramiel have you tested whether `use` works?

**Reply by itsramiel** [reply]
I am not familiar with `use` actually

**Reply by satya164** [reply]
https://react.dev/reference/react/use

**Reply by peterlazar1993** [reply]
Could someone post some examples please. I have been able to use the use api successfully but linter complains about rule of hooks.

**Comment by satya164** [comment]
There's a now a `with` property which lets you pass dynamic props to static config screens https://reactnavigation.org/docs/static-configuration#passing-dynamic-props-or-wrapping-the-navigator
